Unix/doc/setup-kerberos-omi.md

```diff
@@ -0,0 +1,627 @@
+
+## Setting up the client machine
+
+Different Distros and Unix implementations have different tools to accomplish it, but to connect to AD, one must: 
+
+  - Set up automatic time sync with ADC/KDC. This is ussaly via the network time protocol, ntp.
+  - DNS set up so it sees the AD server and is part of the servers DNS domain.
+  - Hostname shorted to 15 characters or less, if needed. Fully qualified hostname by default.
+  - Kerberos and LDAP setup (sssd/realm or winbind and net ads)
+  - nsswitch.conf set to use ldap and sss or winbind
+  - PAM modules added to allow kerberos authentication.
+
+Winbind, which directly uses the samba service, was universally used as an active domain agent on linux until the
+most recent version of Centos/RH, SLES and Ubuntu.  It is being supplanted by the Red Hats
+System Security Services Daemon (sssd) which is more secure and handles identity mapping better.
+
+Winbind uses the utility *net ads* as a command line interface to join a domain, leave the domain, manage keytabs,
+add users and computers, etc.
+
+Both Sssd and winbind use "net ads" to perform a number of useful operations.
+
+MacOS uses its own suite of utilities based on a fork of the Heimdal kerberos implementation (http://www.h5l.org).
+
+Much of the process of preparing a machine to join an active domain is a matter of preparing. IF the server and client 
+machine is already set up right, then the process of joining usually goes very smoothly. If there are problems in the setup, 
+then debugging becomes difficult. This document will try to call out the things to check in advance.
+
+Due to the limitations of netbios which are still reflected in the system, it is a good idea to limit hostnames to 15 characters.
+
+Microsoft recommends this, and we have had some rather unpleasant issues around violating that rule.  The reason is that 
+the netbios host name (referred to in the domain controllers data as SAM is based on the netbios name. Deviation between the shortname and the SAM name can cause problems. It just isnt worth it.
+
+A link here.(https://technet.microsoft.com/en-us/library/cc726016(v=ws.10).aspx)
+
+  - The actual creation of computer and user accounts is done via the "Active Domain Users And Computers" utility on the ADC.
+  - Service Principals are created using the setspn.exe command line utility. 
+  - Information about the active domain state can be gotten from via ldapsearch (Windows or Cleint side) or throgh powershell,
+    via the get-ADComputer() and get-ADUser cmdlets.
+
+### Redhat/Centos Linux 7 (all versions)  (Should also apply to Fedora 24 and later).
+
+Redhat has an excellent write up on setting up an active domain client 
+here.
+(https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/Linux_Domain_Identity_Authentication_and_Policy_Guide/index.html)
+which describes the process with all of the possible considerations.  Some might find it a bit long, but it is recommended
```

Comment:
I'd probably just change this whole thing to read:

Redhat has an excellent write up on setting up an active domain client. [This document](https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/7/html/Linux_Domain_Identity_Authentication_and_Policy_Guide/index.html) describes the process with all of the possible considerations. The document is long, but is recommend reading.